### ISSUE TITLE
HDF5Compound having a HDF5Vlen

### ISSUE THREAD

**Issue by tknopp** [JuliaIO/HDF5.jl#426]
Hi,

as indicated in #425, I would like to read the data from https://sourceforge.net/projects/ismrmrd/files/data/  (e.g. the spiral data) which is an ISMRMRD file (http://onlinelibrary.wiley.com/doi/10.1002/mrm.26089/abstract)

The first issue was fixed in #379. The next issue is that the compound data contains 3 parts:

```
julia> data[1]
HDF5.HDF5Compound{3}((UInt8[0x00,0x00,0x40,0x00,0x00,0x00,0x00,0x00,0x00,0x00  …  0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x00],UInt8[0xd0,0x07,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0xd0,0x1d,0x06,0xbd,0x7f,0x00,0x00],UInt8[0x00,0xfa,0x00,0x00,0x00,0x00,0x00,0x00,0x00,0x70,0x26,0x16,0x01,0x00,0x00,0x00]),("head","traj","data"),(HDF5.HDF5Compound{24},HDF5.HDF5Vlen{Float32},HDF5.HDF5Vlen{Float32}))
```

The first part (header) is just read fine. However, the `HDF5.HDF5Vlen{Float32}` are much too short. The issue seems to be that this sizeof https://github.com/JuliaIO/HDF5.jl/blob/master/src/HDF5.jl#L1446
does not contain the length of the dataset. I am not familiar with HDF5 internals but it seems that here just a "pointer/offset" is stored.

Would be great if someone could help with some more experience. Be browsing HDF5 history it seems that @simonster has quite some insight into HDF5Compound.

Thanks,

Tobi

**Comment by tknopp** [comment]
I said that the header is just read fine but that is not the case. The struct has a sub struct and that seems to be ignored because nested `HDF5Compound` make problems with HDF5.jl.

I was not aware that its possible to store such complex structures within a HDF5 file. Makes it pretty unaccessible IMHO

**Comment by kleinhenz** [comment]
This should be closed by #592.

**Comment by tknopp** [comment]
yes
